src/main/java/edu/hm/hafner/coverage/parser/MetricsParser.java

```diff
@@ -0,0 +1,243 @@
+
+import edu.hm.hafner.coverage.ClassNode;
+import edu.hm.hafner.coverage.CoverageParser;
+import edu.hm.hafner.coverage.CyclomaticComplexity;
+import edu.hm.hafner.coverage.FileNode;
+import edu.hm.hafner.coverage.MethodNode;
+import edu.hm.hafner.coverage.Metric;
+import edu.hm.hafner.coverage.ModuleNode;
+import edu.hm.hafner.coverage.Node;
+import edu.hm.hafner.coverage.PackageNode;
+import edu.hm.hafner.util.FilteredLog;
+import edu.hm.hafner.util.PathUtil;
+import edu.hm.hafner.util.SecureXmlParserFactory;
+import edu.hm.hafner.util.SecureXmlParserFactory.ParsingException;
+import edu.hm.hafner.util.TreeString;
+
+/**
+ * Parses Metrics reports into a hierarchical Java Object Model.
+ *
+ * @author Maximilian Waidelich
+ */
+@SuppressWarnings("PMD.GodClass")
+public class MetricsParser extends CoverageParser {
+    private static final long serialVersionUID = -4461747681863455621L;
+
+    /** XML elements. */
+    private static final QName PACKAGE = new QName("package");
+    private static final QName CLASS = new QName("class");
+    private static final QName METHOD = new QName("method");
+    private static final QName METRIC = new QName("metric");
+    private static final QName FILE = new QName("file");
+
+    /** Required attributes of the XML elements. */
+    private static final QName FQCN = new QName("fqcn");
+    private static final QName NAME = new QName("name");
+    private static final QName BEGIN_LINE = new QName("beginline");
+    private static final QName VALUE = new QName("value");
+
+    private static final String CYCLOMATIC_COMPLEXITY = "CyclomaticComplexity";
+    private static final String COGNITIVE_COMPLEXITY = "CognitiveComplexity";
+    private static final String NCSS = "NCSS";
+    private static final String NPATH_COMPLEXITY = "NPathComplexity";
+
+    private static final PathUtil PATH_UTIL = new PathUtil();
+
+    /**
+     * Creates a new instance of {@link MetricsParser}.
+     */
+    public MetricsParser() {
+        this(ProcessingMode.FAIL_FAST);
+    }
+
+    /**
+     * Creates a new instance of {@link MetricsParser}.
+     *
+     * @param processingMode
+     *         determines whether to ignore errors
+     */
+    public MetricsParser(final ProcessingMode processingMode) {
+        super(processingMode);
+    }
+
+    @Override
+    protected ModuleNode parseReport(final Reader reader, final String fileName, final FilteredLog log) {
+        try {
+            var factory = new SecureXmlParserFactory();
+            var eventReader = factory.createXmlEventReader(reader);
+
+            while (eventReader.hasNext()) {
+                XMLEvent event = eventReader.nextEvent();
+
+                if (event.isStartElement()) {
+                    var startElement = event.asStartElement();
+                    var tagName = startElement.getName();
+                    if (PACKAGE.equals(tagName)) {
+                        var root = new ModuleNode("");
```

Comment:
Can we have a name (or use the `source` attribute) for the `metrics` node that is used as module name? Then we can distinguish multiple reports in the future.